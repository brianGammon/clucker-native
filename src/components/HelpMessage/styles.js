import { StyleSheet } from 'react-native';
import variables from '../../styles/variables';

export default StyleSheet.create({
  container: {
    marginVertical: 10,
    flexDirection: 'row',
    backgroundColor: variables.cluckerOrangeLight,
    borderRadius: 20,
    alignItems: 'center',
  },
  messageContainer: {
    flex: 1,
    marginLeft: 5,
    alignItems: 'center',
  },
  image: {
    width: 50,
    height: 50,
  },
});
