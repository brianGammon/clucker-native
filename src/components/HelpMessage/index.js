import React from 'react';
import { Image } from 'react-native';
import { Text, View } from 'native-base';
import styles from './styles';

type Props = {
  children: React.Component<Text>,
};

const HelpMessage = ({ children }: Props) => (
  <View padder style={styles.container}>
    <Image
      source={require('../../assets/chicken-sm.png')}
      style={styles.image}
    />
    <View style={styles.messageContainer}>{children}</View>
  </View>
);

export default HelpMessage;
