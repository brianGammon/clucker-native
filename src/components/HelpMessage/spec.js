import React from 'react';
import HelpMessage from '.';

describe('HelpMessage', () => {
  test('renders with default message', () => {
    const wrapper = shallow(<HelpMessage />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders with message prop', () => {
    const wrapper = shallow(<HelpMessage message="Passed in message" />);
    expect(wrapper).toMatchSnapshot();
  });
});
