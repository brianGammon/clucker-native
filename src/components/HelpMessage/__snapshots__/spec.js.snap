// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpMessage renders with default message 1`] = `
<Styled(ViewNB)
  padder={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(254, 206, 137)",
      "borderRadius": 20,
      "flexDirection": "row",
      "marginVertical": 10,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/chicken-sm.png",
      }
    }
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  />
  <Styled(ViewNB)
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginLeft": 5,
      }
    }
  />
</Styled(ViewNB)>
`;

exports[`HelpMessage renders with message prop 1`] = `
<Styled(ViewNB)
  padder={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(254, 206, 137)",
      "borderRadius": 20,
      "flexDirection": "row",
      "marginVertical": 10,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/chicken-sm.png",
      }
    }
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  />
  <Styled(ViewNB)
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginLeft": 5,
      }
    }
  />
</Styled(ViewNB)>
`;
